dialer.go

```diff
@@ -120,23 +121,34 @@ func NewDialer(ctx context.Context, opts ...DialerOption) (*Dialer, error) {
 
 // Dial returns a net.Conn connected to the specified Cloud SQL instance. The instance argument must be the
 // instance's connection name, which is in the format "project-name:region:instance-name".
-func (d *Dialer) Dial(ctx context.Context, instance string, opts ...DialOption) (net.Conn, error) {
+func (d *Dialer) Dial(ctx context.Context, instance string, opts ...DialOption) (conn net.Conn, err error) {
+	var endDial trace.EndSpanFunc
+	ctx, endDial = trace.StartSpan(ctx, "cloud.google.com/go/cloudsqlconn.Dial")
+	defer func() { endDial(err) }()
 	cfg := d.defaultDialCfg
 	for _, opt := range opts {
 		opt(&cfg)
 	}
 
+	var endInfo trace.EndSpanFunc
+	ctx, endInfo = trace.StartSpan(ctx, "cloud.google.com/go/cloudsqlconn/internal.InstanceInfo")
 	i, err := d.instance(instance)
 	if err != nil {
+		endInfo(err)
 		return nil, err
 	}
 	addr, tlsCfg, err := i.ConnectInfo(ctx, cfg.ipType)
 	if err != nil {
+		endInfo(err)
 		return nil, err
 	}
-	addr = net.JoinHostPort(addr, serverProxyPort)
+	endInfo(err)
```

Comment:
nit: do you think this would be more readable in an enclosed function?

Reply:
It would be cleaner for the tracing at least. But the result would be a function that took three arguments (context, ipType, and instance)) and returned four (addr, tlsCfg, i. and err). So in this case, it seems better overall to keep these here.